src/server.ts

```diff
@@ -749,3 +753,77 @@ function findCreatedAccountSequenceInTransactionMeta(
 
   throw new Error("No account created in transaction");
 }
+
+
+// TODO - remove once rpc updated to
+// append expiration entry per data LK requested onto server-side response 
+// https://github.com/stellar/soroban-tools/issues/1010
+async function mergeResponseExpirationLedgers(response: Promise<SorobanRpc.RawGetLedgerEntriesResponse>,
+  requestedKeys: xdr.LedgerKey[]
+): Promise<SorobanRpc.RawGetLedgerEntriesResponse> {
+  let ledgerEntriesResponse = await response;
+  const expirationKeyToRawEntryResult:Map<String, SorobanRpc.RawLedgerEntryResult> = new Map();
+  const requestedKeyXdrs = new Set<String>(requestedKeys.map(requestedKey =>
+    requestedKey.toXDR('base64')));
+
+  (ledgerEntriesResponse.entries ?? []).forEach(rawEntryResult => {
+    if (!rawEntryResult.key) {
+      // don't interpret here, just pass it through to the outgoing response as-is
+      expirationKeyToRawEntryResult.set(Math.random().toString(), rawEntryResult)
+      return;
+    }
+    const parsedKey:xdr.LedgerKey = xdr.LedgerKey.fromXDR(rawEntryResult.key, 'base64');
```

Comment:
nit: type is _heavily_ implied here, unneeded verbosity
```suggestion
    const parsedKey = xdr.LedgerKey.fromXDR(rawEntryResult.key, 'base64');
```